pkg/compact/planner.go

```diff
@@ -56,19 +78,17 @@ func (p *tsdbBasedPlanner) Plan(_ context.Context, metasByMinTime []*metadata.Me
 
 // selectMetas returns the dir metas that should be compacted into a single new block.
 // If only a single block range is configured, the result is always nil.
-func selectMetas(ranges []int64, metasByMinTime []*metadata.Meta) []*metadata.Meta {
+func selectMetas(ranges []int64, noCompactMarked map[ulid.ULID]*metadata.NoCompactMark, metasByMinTime []*metadata.Meta) []*metadata.Meta {
```

Comment:
Can you please have a unit test for this function? The logic is not very easy to review...

Reply:
It's copied from Prometheus ): 

Also there are full unit tests on Planner side, which should be enough. WDYT?

I agree code is not very clear. We need to redesign all long term - this is exactly same as Prometheus now

Reply:
Nice